src/averageAges.js

```diff
@@ -55,7 +82,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person =>
+      person.sex === 'm' && people.some(woman =>
+        woman.name === person.mother))
+    : people.filter(person =>
+      people.some(woman => woman.name === person.mother));
+
+  const averageDiff = children.reduce((sum, person) => {
+    const mother = people.find(woman => woman.name === person.mother);
+
+    return sum + (person.born - mother.born);
+  }, 0) / children.length;
```

Comment:
Nice but I suggest to move dividing part out of this constant, it will increase readability